src/utils/mathUtils.ts

```diff
@@ -155,24 +156,31 @@ export function within(p: Point, bounds: Bounds) {
  */
 export function getCumulativeTransform(element: Element): ICssTransformMatrix {
   const elems = getHtmlElementAncestors(element);
-  elems.reverse();
 
   let transform = _IDENTITY_TRANSFORM;
+  let offsetParent: Element = null;
   for (const elem of elems) {
-    // translate coordinates to client origin
-    const offsetX = elem.scrollLeft - elem.offsetLeft - elem.clientLeft;
-    const offsetY = elem.scrollTop - elem.offsetTop - elem.clientTop;
-    transform = premultiplyTranslate([offsetX, offsetY], transform);
-
-    // apply css transform if present
+    // apply css transform from any ancestor element
     const elementTransform = getElementTransform(elem);
     if (elementTransform != null) {
-      const midX = elem.clientWidth/2;
-      const midY = elem.clientHeight/2;
-      transform = premultiplyTranslate([-midX, -midY], transform);
-      transform = multiplyMatrix(invertMatrix(elementTransform), transform);
-      transform = premultiplyTranslate([midX, midY], transform);
+      const midX = elem.clientWidth / 2;
+      const midY = elem.clientHeight / 2;
+      transform = multiplyTranslate(transform, [midX, midY]);
+      transform = multiplyMatrix(transform, invertMatrix(elementTransform));
+      transform = multiplyTranslate(transform, [-midX, -midY]);
```

Comment:
note that we swap the order of arguments and the order of calls because we now iterator over the ancestry starting from the deepest child up to `<body>` (hence the removal of `elems.reverse()` above. we swap the order so we can keep track of the `element.offsetParent`